### PR TITLE
Allow modification of MIPS_BINUTILS_PREFIX

### PR DESCRIPTION
**Pull request by philtyl**

### CODE REVIEW

**Comment by AngheloAlf** [comment]
Could you explain what this PR actually fixes? Currently passing `MIPS_BINUTILS_PREFIX` to make does what we expect
```
$ make MIPS_BINUTILS_PREFIX=thiswillfail-
thiswillfail-as -march=vr4300 -32 -Iinclude data/code/z_play.data.s -o build/data/code/z_play.data.o
make: thiswillfail-as: No such file or directory
make: *** [Makefile:351: build/data/code/z_play.data.o] Error 127
```

**Comment by AngheloAlf** [comment]
I see what you mean, running this again after purging the binutils packages produces an error
```
$ make MIPS_BINUTILS_PREFIX=thiswillfail-
Makefile:69: *** Please install or build mips-linux-gnu.  Stop.
```

**Review by AngheloAlf — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -63,10 +63,9 @@ endif
 
 #### Tools ####
 
-ifeq ($(shell type mips-linux-gnu-ld >/dev/null 2>/dev/null; echo $$?), 0)
-  MIPS_BINUTILS_PREFIX := mips-linux-gnu-
-else
-  $(error Please install or build mips-linux-gnu)
+MIPS_BINUTILS_PREFIX ?= mips-linux-gnu-
```
Comment (hensldm) [inline Makefile:66]: Could you move this to the `Defaults` section? Probably also a good idea to add the comment from OoT's makefile as well.

**Review by hensldm — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]

**Comment by AngheloAlf** [comment]
We are trying to get everybody who has contributed / wants to contribute to MM decomp to agree to licensing the repo before merging PRs from people who has not yet agreed to it. It is better explained [here](https://github.com/louist103/mm-licensing/issues/1). Could you please read about it and post in that issue if you agree or not. Thanks!

**Review by engineer124 — APPROVED** [review]

**Comment by AngheloAlf** [comment]
Friendly reminder that this PR is only waiting for any kind of interaction on the [licensing effort](https://github.com/louist103/mm-licensing/issues/1) before it can actually be merged.

**Comment by AngheloAlf** [comment]
Closed due to inactivity

**Pull request closed without merging**